src/PieChart/components/PieChartContainer.ts

```diff
@@ -0,0 +1,205 @@
+    }
+
+    render() {
+        if (this.props.mxObject) {
+            if (this.state.alertMessage) {
+                return createElement(Alert, {
+                    className: `widget-${this.props.chartType}-chart-alert`,
+                    message: this.state.alertMessage
+                });
+            }
+            return createElement(PieChart, {
+                className: this.props.class,
+                config: {
+                    displayModeBar: this.props.showToolBar
+                },
+                data: [ {
+                    hole: this.props.chartType === "donut" ? .4 : 0,
+                    hoverinfo: "label",
+                    labels: this.state.labels,
+                    marker: {
+                        colors: this.state.colors
+                    },
+                    type: "pie",
+                    values: this.state.values
+                } ],
+                height: this.props.height,
+                heightUnit: this.props.heightUnit,
+                layout: {
+                    autosize: this.props.responsive,
+                    showlegend: this.props.showLegend
+                },
+                style: PieChartContainer.parseStyle(this.props.style),
+                type: this.props.chartType,
+                width: this.props.width,
+                widthUnit: this.props.widthUnit
+            });
+        }
+
+        return null;
+    }
+
+    componentWillReceiveProps(newProps: PieChartContainerProps) {
+        this.resetSubscriptions(newProps.mxObject);
+        this.fetchData(newProps.mxObject);
+    }
+
+    componentWillUnmount() {
+        if (this.subscriptionHandles) {
+            this.subscriptionHandles.forEach(mx.data.unsubscribe);
+        }
+    }
+
+    public static parseStyle(style = ""): { [key: string]: string } {
+        try {
+            return style.split(";").reduce<{ [key: string]: string }>((styleObject, line) => {
+                const pair = line.split(":");
+                if (pair.length === 2) {
+                    const name = pair[0].trim().replace(/(-.)/g, match => match[1].toUpperCase());
+                    styleObject[name] = pair[1].trim();
+                }
+                return styleObject;
+            }, {});
+        } catch (error) {
+            // tslint:disable-next-line no-console
+            console.log("Failed to parse style", style, error);
+        }
+
+        return {};
+    }
+
+    public static validateProps(props: PieChartContainerProps): string {
+        let errorMessage = "";
+        if (props.dataSourceType === "microflow" && !props.dataSourceMicroflow) {
+            errorMessage += ` 'Data source' is set to 'Microflow' but 'Microflow' is missing \n`;
+        }
+
+        return errorMessage && `Configuration error in ${props.chartType}chart:\n\n ${errorMessage}`;
+    }
+
+    private resetSubscriptions(mxObject?: mendix.lib.MxObject) {
+        this.subscriptionHandles.forEach(mx.data.unsubscribe);
```

Comment:
Sooooo, Mr. Akileng, whhhy do we have an array for this subscriptionHandles when it only ever has one item? Why?? :)